osquery/tables/system/windows/ntfs_acl_permissions.cpp

```diff
@@ -0,0 +1,227 @@
+		{ SPECIFIC_RIGHTS_ALL            ,"Specific Rights All" },
+		{ ACCESS_SYSTEM_SECURITY         ,"Access System Security" },
+		{ MAXIMUM_ALLOWED                ,"Maximum Allowed" },
+		{ GENERIC_READ                   ,"Generic Read" },
+		{ GENERIC_WRITE                  ,"Generic Write" },
+		{ GENERIC_EXECUTE                ,"Generic Execute" },
+		{ GENERIC_ALL                    ,"Generic All" }
+	};
+
+	for (auto const& perm : permVals) {
+		if ((pmask & perm.first) != 0)
+			permList.push_back(perm.second);
+	}
+
+	permStr = alg::join(permList, ",");
+	return permStr;
+}
+
+// helper function to convert inheritance type to string
+std::string inheritanceToStr(const unsigned long i) {
+	switch (i) {
+	case CONTAINER_INHERIT_ACE: // equivalent to SUB_CONTAINERS_ONLY_INHERIT
+		return "Container Inherit Ace";
+	case INHERIT_NO_PROPAGATE: // equivalent to NO_PROPAGATE_INHERIT_ACE
+		return "Inherit No Propagate";
+	case INHERIT_ONLY: // equivalent to INHERIT_ONLY_ACE
+		return "Inherit Only";
+	case OBJECT_INHERIT_ACE: // equivalent to SUB_OBJECTS_ONLY_INHERIT
+		return "Object Inherit Ace";
+	case SUB_CONTAINERS_AND_OBJECTS_INHERIT:
+		return "Sub containers and Objects Inherit";
+	case NO_INHERITANCE:
+		return "No Inheritence";
+	default:
+		return "Other";
+	}
+}
+
+// helper function to get account/group name from trustee
+std::string trusteeToStr(const TRUSTEE& t) {
+	char name[256];
+	char domain[256];
+	unsigned long size = 256;
+	//bool r;
+	SID_NAME_USE accountType;
+
+	switch (t.TrusteeForm) {
+	case TRUSTEE_IS_SID:
+		// get the name from the SID
+		if (!LookupAccountSid(NULL, 
+				(PSID) t.ptstrName, name,
+				&size,
+                domain,
+                &size,
+                &accountType)) {
+			//sprintf("LookupAccountSid error: %u\n", GetLastError());
+            TLOG << "LookupAccountSid error: " << GetLastError();
+		}
+		else {
+			return name;
+		}
+	case TRUSTEE_IS_NAME:
+		// get the name from ptstrName
+		return t.ptstrName;
+	case TRUSTEE_BAD_FORM:
+		// invalid
+		return "Invalid";
+	case TRUSTEE_IS_OBJECTS_AND_SID:
+		// ptstrName member is a pointer to an OBJECTS_AND_SID struct
+        if (!LookupAccountSid(
+				NULL,
+                (PSID)((OBJECTS_AND_SID*)t.ptstrName)->pSid,
+                name,
+                &size,
+                domain,
+                &size,
+                &accountType)) {
+			//printf("LookupAccountSid error: %u\n", GetLastError());
+            TLOG << "LookupAccountSid error: " << GetLastError();
+		}
+		else {
+			return name;
+		}
+	case TRUSTEE_IS_OBJECTS_AND_NAME:
+		// ptstrName member is a pointer to an OBJECTS_AND_NAME struct
+		return ((OBJECTS_AND_NAME_*) t.ptstrName)->ptstrName;
+	default:
+		return "";
+	}
+}
+
+QueryData genNTFSACLPerms(QueryContext& context) {
+  QueryData results;
+
+	std::string ObjName;
+	unsigned long result = 0;
+	PACL DACL = NULL;
+	PEXPLICIT_ACCESS aceList = NULL, aceItem;
+	unsigned long aceCount, aceIndex;
```

Comment:
It is much better to not leave variables uninitialised. Prevent us from some stupid mistakes in the future. Just put a zero into it. Or use brace declaration:
```c++
auto aceCount = unsigned{};
auto aceIndex = unsigned{};
```